### ISSUE TITLE
Error Locations

### ISSUE THREAD

**Issue by courtneycorwin** [ombegov/ITDB-schema#324]
Hi All,

I've had inquiries from a few agencies mentioning that without line numbers or object IDs (i.e agencyProjectId, agencyActivityId, etc.) it's difficult to tell what the error is referring to.  Is it possible to add either the line number or object ID back to the error message?

**Comment by jeremiah-tcg** [comment]
Hi Courtney, 
We created ITD-1970 to evaluate the effort for this and added it to our current sprint. I will let you know when we have an ETA for when this will be deployed.
Regards, 
Jeremiah

**Comment by jeremiah-tcg** [comment]
Hi Courtney, 
We deployed 2018.09B to both UAT and production that included a fix for this.  Please let me know if we can close this issue.
Regards, 
Jeremiah

**Comment by courtneycorwin** [comment]
Ah, I see.  That's disappointing - the line numbers have helped out my agencies a lot.

Ok, so in the case of the ITPS, could you maybe include the UII number of the investment that's causing the error?  Provided that it's investment related - I know not all of the errors are.  And then always include the agencyID (or OMBID if that's not available) for the business cases and the SIRs?

**Comment by courtneycorwin** [comment]
What was the fix you all implemented?

**Comment by jeremiah-tcg** [comment]
Hi Courtney, 

We didn't make any changes to IT Budget and Business Case. The IT Budget and Business Case error messages will show line numbers for schema errors and will not show line numbers when the schema is valid. OMB did not want to create schema validations, so more of the field type/length/enumeration type of errors reside in the code for the new submission types. 

Our most recent change primarily focused on SIRs and Systems Inventory Lists. We made the default identifier the agencyID. If agencyID is not found, the OMBID is used if one is available. For a couple of structural/critical errors, UII is used. 

We recognize this may not be the best solution, which is why we created a follow on ticket to evaluate how to do this better and make it more consistent in the future.

Regards, 
Jeremiah
